openerp-client/src/main/java/org/bahmni/openerp/web/http/client/RestClient.java

```diff
@@ -64,21 +68,22 @@ public String post(String URL, String requestBody) {
             WebClient client = getWebClient(baseURL);
             HttpHeaders headers = getHttpHeaders();
             Consumer < HttpHeaders > consumer = httpHeaders -> httpHeaders.addAll(headers);
-            String response = client.post().uri(URL).headers(consumer).cookie("session_id", sessionId).bodyValue(requestBody).retrieve().bodyToMono(String.class).timeout(Duration.ofMillis(connectionTimeout)).block();
+            ResponseEntity<String> responseEntity = client.post().uri(URL).headers(consumer).cookie("session_id", sessionId).bodyValue(requestBody).retrieve().toEntity(String.class).timeout(Duration.ofMillis(connectionTimeout)).block();
+            ResponseChecker.checkResponse(responseEntity);
+            String response = responseEntity.getBody();
             logger.debug("\n-----------------------------------------------------{} Initiated-----------------------------------------------------\n* Cookies : {}\n* Request : {}\n* Response : {}\n-----------------------------------------------------End of {}-----------------------------------------------------", URL, sessionId, requestBody, response, URL);
-            if (response == null) {
-                throw new OpenERPException(String.format("Could not post to %s", URL));
-            }
-            logger.debug("Post Data output: {}", response);
             logger.debug("Post Data output: {}", response);
             return response;
+        } catch (OdooRestException e) {
+            logger.error("Could not post to {}", URL, e);
```

Comment:
Since OdooRestException is thrown after making the post call, the log message and exception message could be `Post call to URL failed`

Reply:
@mohan-13 done.